standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -182,8 +182,11 @@ static void cleanupHandlerContext() {
       HMSHandlerContext.getHMSHandler().ifPresent(handler -> {
         handler.notifyMetaListenersOnShutDown();
       });
-      HMSHandlerContext.clear();
-      logAndAudit("Done cleaning up thread local RawStore");
+      HMSHandlerContext.getRawStore().ifPresent(rs -> {
+        logAndAudit("Done cleaning up thread local RawStore");
```

Comment:
This log should be printed after `HMSHandlerContext.clear()`, Once the clean up is done.

Reply:
I also had this discomfort before. What change my mind is that the audit log is related to only RawStore, and it was cleaned up here: https://github.com/apache/hive/blob/7bfc54ffc8baf5f3e7de326b750fe9355b11132b/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L178-L179

So I accepted this change, because `HMSHandlerContex.clear()` is for context rather than RawStore. Do you think we need reorder the log?

Reply:
Thanks for the clarification, can you please explain the need of `HMSHandlerContext.getRawStore().ifPresent` here 
I dont see `rs` being used here, is there a better way to verify if the Raw Store is cleared or not?